src/Bicep.Core/Semantics/ArmTemplateSemanticModel.cs

```diff
@@ -135,32 +140,276 @@ public bool HasErrors()
             return diagnosticWriter.GetDiagnostics().Count > 0;
         }
 
-        private static TypeSymbol GetType(TemplateInputParameter parameter)
+        private TypeSymbol GetType(TemplateInputParameter parameter) => parameter.Type?.Value switch
         {
-            var allowedValueTypes = GetAllowedValueTypes(parameter);
+            TemplateParameterType.String when TryCreateUnboundResourceTypeParameter(GetMetadata(parameter), out var resourceType) =>
+                resourceType,
 
-            bool AllowedStringLiteralsProvided() => allowedValueTypes.Any() && allowedValueTypes.All(x => x is StringLiteralType);
+            _ => GetType((ITemplateSchemaNode) parameter),
+        };
 
-            return parameter.Type.Value switch
+        private TypeSymbol GetType(ITemplateSchemaNode schemaNode)
+        {
+            if (schemaNode.Ref?.Value is string @ref)
```

Comment:
Apologies, not familiar with `@ref` and couldn't find anything on it. Is this just a fancy C# way of passing a reference?

Reply:
`@ref` is just the variable name here. (The at sign is needed because `ref` is a language keyword in C#.)

This line is just making sure that `schemaNode.Ref?.Value` is a non-null string, then assigning it to the `@ref` variable to be used within the if block.

Reply:
Ah, that makes a lot more sense. Simpler than I thought haha